airflow/contrib/operators/gcp_spanner_operator.py

```diff
@@ -197,3 +200,197 @@ def execute(self, context):
+                                   "'{}' in project '{}' and instance '{}' is missing."
+                                   " Create the database first before you can update it.".
+                                   format(self.database_id,
+                                          self.project_id,
+                                          self.instance_id))
+        else:
+            return self._hook.update_database(project_id=self.project_id,
+                                              instance_id=self.instance_id,
+                                              database_id=self.database_id,
+                                              ddl_statements=self.ddl_statements,
+                                              operation_id=self.operation_id)
+
+
+class CloudSpannerInstanceDatabaseDeleteOperator(BaseOperator):
+    """
+    Deletes a Cloud Spanner database.
+
+    :param project_id: The ID of the project that owns the Cloud Spanner Database.
+    :type project_id: str
+    :param instance_id: Cloud Spanner instance ID.
+    :type instance_id: str
+    :param database_id: Cloud Spanner database ID.
+    :type database_id: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    """
+    # [START gcp_spanner_database_delete_template_fields]
+    template_fields = ('project_id', 'instance_id', 'database_id',
+                       'gcp_conn_id')
+    # [END gcp_spanner_database_delete_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 instance_id,
+                 database_id,
+                 gcp_conn_id='google_cloud_default',
+                 *args, **kwargs):
+        self.instance_id = instance_id
+        self.project_id = project_id
+        self.database_id = database_id
+        self.gcp_conn_id = gcp_conn_id
+        self._validate_inputs()
+        self._hook = CloudSpannerHook(gcp_conn_id=gcp_conn_id)
+        super(CloudSpannerInstanceDatabaseDeleteOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is empty")
+        if not self.instance_id:
+            raise AirflowException("The required parameter 'instance_id' is empty")
+        if not self.database_id:
+            raise AirflowException("The required parameter 'database_id' is empty")
+
+    def execute(self, context):
+        db = self._hook.get_database(self.project_id,
+                                     self.instance_id,
+                                     self.database_id)
+        if not db:
+            self.log.info("The Cloud Spanner database was missing: "
+                          "'{}' in project '{}' and instance '{}'. Assuming success.".
+                          format(self.database_id, self.project_id, self.instance_id))
```

Comment:
Update to `%s` format

Reply:
Sorry for the rework. Fixed all of those.